docs/embed.md

```diff
@@ -0,0 +1,196 @@
+# Embedding
+
+Embedding allows to include parts of adhocracy on other websites. A common use
+case would be the website of a municipal administration that embeds an idea
+collection.  This allows users to access (and eventually contribute to) the
+participation right where they are instead of going to a third party platform.
+
+## TL;DR
+
+You can embed any project with the following code:
+
+    <iframe src="http://example.com/embed/projects/{slug}/" />
+
+## Terms
+
+-   iframe: The HTML element that enables embedding. It is basically a virtual
+    screen that is completely isolated from the embedding website for security
+    reasons.
+-   platform: The regular, non-embedded version of adhocracy. Some
+    functionality is only available on the platform.
+
+## Inspiration and Approaches
+
+While working on this concept we looked at several existing embedding systems
+and found three distinct categories:
+
+-   *Widgets* with limited interaction (e.g. [twitter timeline](https://dev.twitter.com/web/embedded-timelines))
+-   *Applications* (e.g. [soundcloud player](https://help.soundcloud.com/hc/en-us/articles/115003449627-The-HTML5-embedded-player))
+-   Functionality that *integrates* into the website (e.g. [disqus](https://disqus.com/))
+
+## Our chosen approach
+
+It was hard to decide between application and integration, but we finally
+landed on application.
+
+We embed fully functional projects. Anything outside of a project (e.g.
+managing projects, changing account settings) happens on the platform.
+
+In the iframe, we load a JavaScript application called the *embed shell* that
+loads the actual contents via ajax.  Whenever a link is clicked or a form is
+submitted, the corresponding event is intercepted, the request is made via ajax
+and the response is put back into the DOM.
+
+The behavior of links and forms can be modified by adding a `data-embed-target`
+attribute. Possible values are:
+
+-   internal (default): use the mechanism described above
+-   external: open in top frame
+-   ignore: do not intercept the event (useful if other event handlers are
+    registered for this element)
+
+## Restrictions on the overall application
+
+The embed shell uses the regular views. This way, we do not need to duplicate
+all views. However, this also means that you need to be aware of some general
+restrictions:
+
+-   Add `data-embed-target="external"` to all links and forms that should not
+    open inside of the iframe.
+-   Do not trigger navigation from JavaScript
+-   Do not add submit handlers to forms that do not have
+    `data-embed-target="ignore"`.
```

Comment:
Sometimes you do not want to stop event propagation. Or the event listener uses event delegation itself, in which case it is not clear which one will be executed first.

In my first draft the formulation was more like "be careful when …". I changed it to be more specific. This does not mean that you must not ever do this, it is just a recommendation to avoid trouble.